activesupport/lib/active_support/cache.rb

```diff
@@ -806,17 +894,19 @@ def dump(entry)
     class Entry # :nodoc:
       attr_reader :version
 
-      DEFAULT_COMPRESS_LIMIT = 1.kilobyte
+      class << self
+        def unpack(members)
+          new(members[0], expires_at: members[1], version: members[2])
+        end
+      end
 
       # Creates a new cache entry for the specified value. Options supported are
-      # +:compress+, +:compress_threshold+, +:version+, +:expires_at+ and +:expires_in+.
-      def initialize(value, compress: true, compress_threshold: DEFAULT_COMPRESS_LIMIT, version: nil, expires_in: nil, expires_at: nil, **)
+      # +:version+, +:expires_at+ and +:expires_in+.
+      def initialize(value, version: nil, expires_in: nil, expires_at: nil, **)
```

Comment:
`Entry#initialize` no longer automatically compress. The coder can now call `compressed` on it if it needs it, but overall this makes the entry mostly a simpler object.